cmake/ElektraCompiling.cmake

```diff
@@ -84,9 +84,9 @@ set (COMMON_FLAGS "${COMMON_FLAGS} -Wsign-compare -Wfloat-equal")
 set (COMMON_FLAGS "${COMMON_FLAGS} -Wformat-security")
 set (COMMON_FLAGS "${COMMON_FLAGS} -Wshadow")
 set (COMMON_FLAGS "${COMMON_FLAGS} -Wcomments -Wtrigraphs -Wundef")
-set (COMMON_FLAGS "${COMMON_FLAGS} -Wuninitialized -Winit-self -Wmaybe-uninitialized")
```

Comment:
Clang only understands `-Wuninitialized` on OS X. Will use the `APPLE` variable to check for OS  X.